### PR TITLE
feat(ui): Add stability chart to project detail

### PR DESCRIPTION
**Pull request by matejminar**
This is the first chart ever that takes data from the brand new sessions API.

Once we start adding more session charts and things become more clear, I'll start extracting more functions to reusable bits to make working with sessions API easier.

image

Project detail is still behind an internal feature flag.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectDetail/charts/sessionsRequest.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            const totalIntervalSessions = responseData.groups.reduce(
+              (acc, group) =>
+                acc + group.series['sum(session)'].slice(0, dataMiddleIndex)[i],
+              0
+            );
+
+            const intervalCrashedSessions =
+              responseData.groups
+                .find(group => group.by['session.status'] === 'crashed')
+                ?.series['sum(session)'].slice(0, dataMiddleIndex)[i] ?? 0;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/projectDetail/charts/sessionsRequest.tsx:240]: These calculations are kind of a mess right now and share a lot of code with the current timeseries data. I'll think of a better abstraction as we add more session charts.

File: src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  get legend() {
+    const {theme} = this.props;
+
+    return {
+      right: 10,
+      top: 0,
+      icon: 'circle',
+      itemHeight: 8,
+      itemWidth: 8,
+      itemGap: 12,
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx:169]:
```suggestion
      itemGap: space(1.5),
```

File: src/sentry/static/sentry/app/views/releases/utils/sessionTerm.tsx [hunk 3]
```diff
@@ -28,6 +29,7 @@ export const commonTermsDescription = {
   [SessionTerm.CRASHES]: t('Number of sessions with a crashed state'),
   [SessionTerm.CRASH_FREE_USERS]: t('Number of unique users with non-crashed sessions'),
   [SessionTerm.CRASH_FREE_SESSIONS]: t('Number of non-crashed sessions'),
+  [SessionTerm.STABILITY]: t('Stability is the percentage of crash free sessions.'),
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/releases/utils/sessionTerm.tsx:32]: should we remove the punctuation here?

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions, but it looks good 😉

File: src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      align: 'left' as const,
+      textStyle: {
+        color: theme.textColor,
+        verticalAlign: 'top',
+        fontSize: 11,
```
[the remark this answers is not in the document]
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx:174]: I tried but unfortunately it needs to be a number

File: src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        fontFamily: 'Rubik',
```
[the remark this answers is not in the document]
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx]: This one works, I replaced it 👍

File: src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/charts/projectStabilityChart.tsx:169]: yep, needs to be a number

File: src/sentry/static/sentry/app/views/releases/utils/sessionTerm.tsx (hunk 3 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/releases/utils/sessionTerm.tsx:32]: I wanted to be consistent as other terms on that page (coming from performance) have dots

**Pull request merged**